### DISCUSSION TITLE
Labeler workflow requires read & write workflow permissions

### DISCUSSION THREAD

**Discussion by Haasie** [onedr0p/cluster-template#1138] [Q&A] [answered]
After the first updates provided by renovate bot i was experiencing errors with the Meta labeler  action on GH.
The error said "Resource not accessible by integration". 

After some searched it seems that you need to provide read/write access for those actions in order to do there "job" 

Maybe worth mentioning in the Docs :smile:

<img width="765" alt="image" src="https://user-images.githubusercontent.com/45860400/213405521-419fc7e1-95e0-4146-a193-29abcbf50993.png">

**Comment by onedr0p** [comment]
I believe this was done in https://github.com/onedr0p/flux-cluster-template/commit/8875a8b7069443b32c4d3a3eb121c1ea4c244916

**Comment by giannoug** [comment]
This doesn't automatically work, I cloned the repo yesterday and had to apply the permission fix myself to get the Action working 🤔 It failed until I gave it "Read and write permissions", which by default had "Read repository contents and packages permissions"

**Comment by onedr0p** [comment]
Interesting, I'll reopen the issue and hopefully someone can provide a better fix.

**Comment by onedr0p** [comment]
I just made [this change](https://github.com/onedr0p/flux-cluster-template/commit/929fdbd1b512ab41435cc9386af751179b8358fd) which is documented [here](https://github.com/actions/labeler#create-workflow). If this doesn't fix it I am unsure why it is failing.

**Comment by giannoug** [comment]
Unfortunately it didn't work on my newly created (test) repo: https://github.com/giannoug/test/actions/runs/7171702379 (repo will be deleted soon, posting the error below for future reference)

```
✗ {"message":"Resource not accessible by integration","statusCode":403,"headers":{"server":"GitHub.com","date":"Mon, 11 Dec 2023 18:21:55 GMT","content-type":"application/json; charset=utf-8","content-length":"132","x-github-media-type":"github.v3; param=symmetra-preview; format=json","x-accepted-github-permissions":"issues=write; pull_requests=write","x-github-api-version-selected":"2022-11-28","x-ratelimit-limit":"1000","x-ratelimit-remaining":"996","x-ratelimit-reset":"1702322515","x-ratelimit-used":"4","x-ratelimit-resource":"core","access-control-expose-headers":"ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset","access-control-allow-origin":"*","strict-transport-security":"max-age=31536000; includeSubdomains; preload","x-frame-options":"deny","x-content-type-options":"nosniff","x-xss-protection":"0","referrer-policy":"origin-when-cross-origin, strict-origin-when-cross-origin","content-security-policy":"default-src 'none'","vary":"Accept-Encoding, Accept, X-Requested-With","x-github-request-id":"6C40:7CDD:60227:C74C1:65775343","connection":"close"},"body":{"message":"Resource not accessible by integration","documentation_url":"https://docs.github.com/rest/issues/labels#create-a-label"},"method":"POST","endpoint":"/repos/giannoug/test/labels"}
```

**Comment by onedr0p** [comment]
That's unfortunate, I'm not sure how to solve this without clicking thru that config :/

**Comment by gavinmcfall** [comment, marked as answer]
If anyone else comes across this issue and is not actually sure how to fix it.

1. In Github, go into your Repository
2. Click "Settings" in the top menu
3. In the new left menu scroll down until you see "Actions" and expand that menu
4. Click "General"
5. Scroll down under these settings until you find: "Workflow permissions"
6. Set Permissions to "Read and Write Permissions" and click "Save".
7. In the top menu, Click "Actions"
8. In the left Menu click "Label Sync"
9. On the right side of the screen click the drop down "Run Workflow"
10. Click the blue button "Run Workflow"
11. After a few seconds you should see a new run of this action and it should pass

**Reply by iSevenDays** [reply]
Still actual in 2025! Thank you very much! These steps should be in Readme file
